packages/core/src/conflict-resolution.ts

```diff
@@ -271,24 +267,21 @@ export class ConflictResolution {
     const itr = log.entries()
     let entry = itr.next()
     while (!entry.done) {
-      const logEntry = entry.value[1]
-      const commit = await this.dispatcher.retrieveCommit(logEntry.cid)
+      const logEntry = await this.expandLogEntry(entry.value[1])
+      const commitMeta = { cid: logEntry.cid, timestamp: logEntry.timestamp }
       // TODO - should catch potential thrown error here
 
-      let payload = commit
-      if (StreamUtils.isSignedCommit(commit)) {
-        payload = await this.dispatcher.retrieveCommit(commit.link)
-      }
-
-      if (payload.proof) {
-        // it's an anchor commit
+      if (logEntry.commit.proof) {
+        // It's an anchor commit
         // TODO: Anchor validation should be done by the StreamHandler as part of applying the anchor commit
-        await verifyAnchorCommit(this.dispatcher, this.anchorValidator, commit)
-        state = await handler.applyCommit(commit, logEntry, this.context, state)
+        await verifyAnchorCommit(this.dispatcher, this.anchorValidator, logEntry.commit)
+        state = await handler.applyCommit(logEntry.commit, commitMeta, this.context, state)
```

Comment:
Maybe we should modify `applyCommit` to take an logEntry object?
Like so:
```suggestion
        state = await handler.applyCommit(logEntry, this.context, state)
```

Reply:
That was definitely one of the things I'd done in a previous iteration of the code before I trimmed it down. My thinking here was to avoid changing any interfaces and increasing the scope of the changes, since more code and tests would be affected. I plan to open a new ticket to update the interface because the IPFS lookup inside `applyCommit` can totally be removed too.

Reply:
cc @ukstv @stbrody

Reply:
Yeah, I think it's fine to do this in a follow-up PR, but should definitely be done.

Reply:
Opened #1532 and added it to the backlog.